railties/lib/rails/application.rb

```diff
@@ -228,11 +228,20 @@ def config_for(name, env: Rails.env)
 
       if yaml.exist?
         require "erb"
-        config = YAML.load(ERB.new(yaml.read).result) || {}
-        config = (config["shared"] || {}).merge(config[env] || {})
+        all_configs = YAML.load(ERB.new(yaml.read).result) || {}
 
-        ActiveSupport::OrderedOptions.new.tap do |options|
-          options.update(NonSymbolAccessDeprecatedHash.new(config))
+        config = all_configs[env]
+        if shared = all_configs["shared"]
+          config = config ? shared.merge(config) : shared
```

Comment:
If `config` is not a hash and there is `shared` would not this `merge` fail as well?

Reply:
Yes, the assumption here is that you wouldn't set a `shared` key of the values are not hashes.

But I can do a respond_to? check if you think it makes sense.

Reply:
How about do the merge inside the `if config.is_a?(Hash)` branch?

Reply:
That's what I meant by `respond_to?` check.

Reply:
Yeah, I think it is better to move there.